types/tsconfig.json

```diff
@@ -6,7 +6,6 @@
     "noImplicitThis": true,
     "strictNullChecks": true,
     "strictFunctionTypes": true,
-    "esModuleInterop": true,
```

Comment:
Would documenting that `esModuleInterop` should be enabled, resolve this issue without reverting the typings refactor?

Reply:
The `esModuleInterop` setting should only be used in end applications, not libraries. It's a way for app developers to treat CommonJS modules as though they were ESM modules, but it results in lots of little helper functions being injected into the code to handle the mapping between the two module formats. App developers should be able to opt-in or opt-out of that behavior, depending on whether they're okay with the trade-off.